Services/Authentication/classes/class.ilSessionDBHandler.php

```diff
@@ -1,119 +1,90 @@
-    *										eg. mysql://phpsessmgr:topsecret@db.example.com/sessiondb
-    * @param	string		$name			session name [PHPSESSID]
-    */
-    public function open($save_path, $name)
+
+    /**
+     * Opens session, normally a db connection would be opened here, but
+     * we use the standard ilias db connection, so nothing must be done here
+     * @param string $path
+     * @param string $name session name [PHPSESSID]
+     */
+    public function open($path, $name) : bool
     {
         return true;
     }
 
     /**
-    * close session
-    *
-    * for a db nothing has to be done here
-    */
-    public function close()
+     * close session
+     *
+     * for a db nothing has to be done here
+     */
+    public function close() : bool
     {
         return true;
     }
 
-    /*
-    * Reads data of the session identified by $session_id and returns it as a
-    * serialised string. If there is no session with this ID an empty string is
-    * returned
-    *
-    * @param	integer		$session_id		session id
-    */
-    public function read($session_id)
+    /**
+     * Reads data of the session identified by $session_id and returns it as a
+     * serialised string. If there is no session with this ID an empty string is
+     * returned
+     * @param string $id
+     */
+    public function read($id) : string
     {
-        return ilSession::_getData($session_id);
+        return ilSession::_getData($id);
     }
 
     /**
-    * Writes serialized session data to the database.
-    *
-    * @param	integer		$session_id		session id
-    * @param	string		$data			session data
-    */
-    public function write($session_id, $data)
+     * Writes serialized session data to the database.
+     * @param string $id session id
+     * @param string $data session data
+     */
+    public function write($id, $data) : bool
     {
-        $cwd = getcwd();
         chdir(IL_INITIAL_WD);
-        include_once("./Services/Authentication/classes/class.ilSession.php");
-        $r = ilSession::_writeData($session_id, $data);
-        // see bug http://www.ilias.de/mantis/view.php?id=18000
-        //chdir($cwd);
-        return $r;
+
+        return ilSession::_writeData($id, $data);
     }
 
     /**
-    * destroy session
-    *
-    * @param	integer		$session_id			session id
-    */
-    public function destroy($session_id)
+     * Destroys session
+     * @param string $id session id
+     */
+    public function destroy($id) : bool
     {
-        return ilSession::_destroy($session_id);
+        return ilSession::_destroy($id);
     }
 
     /**
-    * removes sessions that weren't updated for more than gc_maxlifetime seconds
-    *
-    * @param	integer		$gc_maxlifetime			max lifetime in seconds
-    */
-    public function gc($gc_maxlifetime)
+     * Removes sessions that weren't updated for more than gc_maxlifetime seconds
+     * @param int $max_lifetime Maximum lifetime in seconds
+     */
+    public function gc($max_lifetime) : bool
```

Comment:
The interface demands  the return type: int | bool

Reply:
Changed with the latest commit